### PR TITLE
feat : Implement seller product creation functionality

### PR DESCRIPTION
**Pull request by Aimeana100**
#a21b3747e94bbcc55a0c5785601f19f99bf56e6b This commit adds a new product to the vendor's collection

This commit includes the following changes:

-  created 4 models `category`, `product`, `productAttribute` and `productImage`
-  created the routes to CRUD categories that should be accessed by admin only
-  created the routes to add new product *only vendor will add product in his / her own collection *
-  created the routes to add product attributes and images as well in approprite model
-  handle responses accordingly
-  create swagger doc for the routes

## How to test 

1. Clone the project from branch `ft-create-new-product`
2.  Copy `.env.example` to `.env` and fill the variable with actual and real values accordingly 
3.  || in case you run docker   do `docker compose up ` on project root directory or 
4. Run `npm i` to install the dependancy
5. Run `npm run dev` to start the server
6. Open Local IP or localhost in browser at port the server started to `ie : 5000` or 4000
7. Test the app using swangger  
8. registe as an admin, create categories , .. 
9. . create a user vendor and login again as vendor
10. .  create a product , ...

### CODE REVIEW

**Review by Evaristekanova — APPROVED** [review]
@BenoitIT  and @Aimeana100  resolve conflicts in readme file

**Review by Evaristekanova — CHANGES_REQUESTED** [review]
resolve conflict from readme file

**Review by BenoitIT — APPROVED** [review]

**Review by FabCode67 — APPROVED** [review]

**Review by Kevin-ishimwe — APPROVED** [review]
great work

File: jest.config.js [hunk 1]
```diff
@@ -16,6 +16,14 @@ module.exports = {
     '!src/utils/sendEmail.js',
     '!src/docs/**/*.js',
     "!src/controllers/Auth/google_oauth.js",
+    "!src/controllers/Auth/updateUser.js",
+    "!src/controllers/products/productController.js",
+    "!src/controllers/products/categoryController.js",
```
Comment (mansurissa) [inline jest.config.js:21]: Create issues to test all testable files here

**Review by mansurissa — APPROVED** [review]
Merging this to unblock others but make sure you write test for everything

**Comment by Aimeana100** [comment]
[A seller should be able to Create/Add a product](https://trello.com/c/8VAiMVMC/15-a-seller-should-be-able-to-create-add-a-product)

**Pull request merged**